trade-finance-manager-api/src/v1/services/deal-cancellation/deal-cancellation.service.ts

```diff
@@ -0,0 +1,58 @@
+import { endOfDay, format, isAfter, toDate } from 'date-fns';
+import { TEAMS, TfmDealCancellation, TfmFacility } from '@ukef/dtfs2-common';
+import sendTfmEmail from '../send-tfm-email';
+import { CANCEL_DEAL_PAST_DATE, CANCEL_DEAL_FUTURE_DATE } from '../../../constants/email-template-ids';
+import * as api from '../../api';
+import { formatFacilityIds } from './helpers/format-facility-ids';
+
+export class DealCancellationService {
+  /**
+   * Sends email to PIM email address confirming the cancellation has been submitted
+   * @param ukefDealId ukef deal id
+   * @param dealCancellation deal cancellation object
+   * @param facilityIds ukef facility ids
+   */
+  private static async sendDealCancellationEmail(ukefDealId: string, dealCancellation: TfmDealCancellation, facilityIds: string[]) {
+    const effectiveFromDate = toDate(dealCancellation.effectiveFrom);
+    const endOfToday = endOfDay(new Date());
+
+    const { email: pimEmail } = await api.findOneTeam(TEAMS.PIM.id);
+
+    const emailTemplateId = isAfter(effectiveFromDate, endOfToday) ? CANCEL_DEAL_FUTURE_DATE : CANCEL_DEAL_PAST_DATE;
+
+    await sendTfmEmail(emailTemplateId, pimEmail, {
+      ukefDealId,
+      effectiveFromDate: format(dealCancellation.effectiveFrom, 'd MMMM yyyy'),
+      bankRequestDate: format(dealCancellation.bankRequestDate, 'd MMMM yyyy'),
+      cancelReason: dealCancellation.reason || '-',
+      formattedFacilitiesList: formatFacilityIds(facilityIds),
+    });
+  }
+
+  /**
+   * Maps the facilities objects into their UKEF ids
+   * @param facilities Facilities
+   * @returns UKEF facility Ids
+   */
+  private static getFacilityIds(facilities: TfmFacility[]) {
+    return facilities.map((facility) => facility.facilitySnapshot.ukefFacilityId).filter((id) => id !== null);
```

Comment:
If the ID is one of these the email won't be sent - is that the behaviour you'd want?

Reply:
Yes please to avoid email being sent out, this is an extremely edge case but if in case deal does makes it way to TFM with those IDs then cancellation should not send email with those IDs. Thanks

Reply:
Should the email still be sent? Thats the current behaviour as written

Reply:
I would not expect an email to be sent to the user with facility IDs as `PENDING` and/or `00000000`.

Reply:
If one facility has ID `PENDING` and another is valid, should the email be sent with only one ID?

Reply:
Have changed the getUkefFacilityIds function & throw an error now if any facility has one of these Id